### PR TITLE
[Access] Added collection_id and block_id to GetTransactionResult arguments

### PR DESCRIPTION
**Pull request by Guitarheroua**
https://github.com/onflow/flow-go/issues/2099

### Context

This pull request adds two new optional arguments to GetTransactionResult: collection_id and block_id. These arguments will help avoid duplicate transaction hashes. If both arguments are missing, the default lookup by transaction ID is performed. The block ID lookup has the highest priority, followed by the collection ID lookup if they are passed. This change ensures backward compatibility with previous versions of the code.

As part of this PR:

The protobuf and Open API were extended(https://github.com/onflow/flow/pull/1335)

### CODE REVIEW

File: engine/access/rpc/backend/backend_transactions.go [hunk 1]
```diff
@@ -234,6 +234,8 @@ func (b *backendTransactions) GetTransactionsByBlockID(
 func (b *backendTransactions) GetTransactionResult(
 	ctx context.Context,
 	txID flow.Identifier,
+	blockID flow.Identifier,
+	collectionID flow.Identifier,
```
Comment (peterargue) [inline engine/access/rpc/backend/backend_transactions.go:238]: I think the logic in this method needs to be tweaked a bit to ensure that any response matches the search criteria exactly. There seems to be 4 cases:

TxID
* Lookup as it's done today
* Query the EN for the result using the discovered block
* Get the collectionID from the block for the response

TxID, BlockID
* Lookup the block by ID
* Query the EN for the result using the provided block
* Get the collectionID from the block for the response

TxID, CollectionID
* Lookup block by collection ID
* Lookup the collection and make sure the tx exists in it
* Query the EN for the result using the discovered block
* Use the provided collectionID in response

TxID, BlockID, CollectionID
* Lookup the block by ID
* Make sure the block contains the provided collection
* Lookup the collection and make sure the tx exists in it
* Query the EN for the result using the provided block
* Use the provided collectionID in response

In any case where the txID/collectionID/blockID don't match eachother, we could return a `NotFound` with some context about which check failed

**Review by peterargue — COMMENTED** [review]
Thanks @Guitarheroua! It could use some more input validation but it's pretty close.

File: engine/access/access_test.go [hunk 2]
```diff
@@ -696,6 +696,174 @@ func (suite *Suite) TestGetSealedTransaction() {
...
+		// Test behaviour with collectionId provided
+		suite.Run("Get transaction result by collection ID", func() {
+			getReq := &accessproto.GetTransactionRequest{
+				Id:           txId[:],
+				CollectionId: collectionId[:],
+			}
+			resp, err := handler.GetTransactionResult(context.Background(), getReq)
+			assertTransactionResult(resp, err)
+		})
+	})
```
Comment (peterargue) [inline engine/access/access_test.go:864]: please also add a test with both `BlockId` and `CollectionId` as well as negative tests for each of the 4 cases

**Review by peterargue — COMMENTED** [review]
Thanks for making those changes. A few more comments, but it's very close.

**Review by peterargue — APPROVED** [review]
Looks good once the conflicts are addressed and the imports is switched back to `onflow/flow`

**Review by koko1123 — APPROVED** [review]
looks good to me! aside from the dependency change in `go.mod`

**Comment by peterargue** [comment]
> > looks good to me! aside from the dependency change in `go.mod`
> 
> @peterargue @koko1123 This one will be blocked if [1335](https://github.com/onflow/flow/pull/1335) will not be reviewed. I do not know who is responsible for this repo to review.

ah, sorry. I missed that one. Just approved. I'll work on getting it merged

**Comment by peterargue** [comment]
@Guitarheroua the protobuf changes were just merged

**Pull request merged**